homeassistant/components/media_player/itunes.py

```diff
@@ -438,13 +438,13 @@ def set_volume_level(self, volume):
     def turn_on(self):
         """Select AirPlay."""
         self.update_state({"selected": True})
-        self.update_ha_state()
+        self.schedule_update_ha_state()
```

Comment:
This one looks risky. I think it either needs to stay or be moved to the bottom of the method.

Also, if we're in clean-up mode, this entity has an empty `update` method. We should switch it to should_poll false.